### PR TITLE
Fix HostName logging for invalid object names during MmiSet

### PR DESCRIPTION
**Pull request by simonjaeger**
## Description

Fixing one of the logging outputs of the `HostName` module. If `MmiSet` is called with an invalid object name, the logging output can be misleading.

Actual:
`[2022-01-05 16:52:03] [HostNameBase.cpp:90] [ERROR] Set called with an invalid object name: 'INVALID_OBJECT_NAME' (expected 'Name' or 'Hosts')`

Expected:
`[2022-01-05 16:52:03] [HostNameBase.cpp:90] [ERROR] Set called with an invalid object name: 'INVALID_OBJECT_NAME' (expected 'DesiredName' or 'DesiredHosts')`

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by robertschaedler3 — DISMISSED** [review]

File: src/modules/hostname/src/lib/HostNameBase.cpp [hunk 1]
```diff
@@ -87,7 +87,7 @@ int HostNameBase::Set(
 
     if (!IsValidObjectName(objectName, true))
     {
-        OsConfigLogError(HostNameLog::Get(), ERROR_INVALID_OBJECT, "Set", !objectName ? "-" : objectName, g_propertyName, g_propertyHosts);
+        OsConfigLogError(HostNameLog::Get(), ERROR_INVALID_OBJECT, "Set", !objectName ? "-" : objectName, g_propertyDesiredName, g_propertyDesiredHosts);
```
Comment (MariusNi) [inline src/modules/hostname/src/lib/HostNameBase.cpp:90]: >!objectName ? "-" : objectName [](http://example.com/codeflow?start=74&length=30)

nit: why not reversed, would look simpler overall

Comment (MariusNi) [reply src/modules/hostname/src/lib/HostNameBase.cpp:90]: objectName ? objectName : "-"

**Review by MariusNi — DISMISSED** [review]
:shipit:

File: src/modules/hostname/src/lib/HostNameBase.cpp (hunk 1 above)
Comment (simonjaeger) [reply src/modules/hostname/src/lib/HostNameBase.cpp]: Fixed, sorry for the delay.

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by MariusNi — APPROVED** [review]

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**